### PR TITLE
[Distributed] [6/N] Fix clang-tidy warnings in torch/csrc/distributed/c10d 

### PR DESCRIPTION
**Pull request by cyyever**
This PR continues to fix some clang-tidy warnings in distributed/c10d code, following https://github.com/pytorch/pytorch/pull/124043.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k

### CODE REVIEW

File: torch/csrc/distributed/c10d/reducer.hpp [hunk 1]
```diff
@@ -538,7 +538,7 @@ class TORCH_API Reducer {
   std::unordered_map<size_t, std::string> param_names_;
   // Variable indices stored sequentially in order of when the gradient is ready
   // for the current backwards pass.
-  std::vector<int> grad_ready_order_indices_;
+  std::vector<size_t> grad_ready_order_indices_;
```
Comment (ezyang) [inline torch/csrc/distributed/c10d/reducer.hpp:541]: verified signed to unsigned seems to be ok

File: torch/csrc/distributed/c10d/Utils.hpp [hunk 2]
```diff
@@ -280,57 +280,57 @@ inline void assertLayoutMatch(
...
 }
 
 inline void assertRootRank(
-    std::function<void(const std::string&)> fn,
-    int rank,
-    int size) {
-  if (rank < 0 || rank >= size) {
+    const std::function<void(const std::string&)>& fn,
+    int64_t rank,
+    size_t size) {
```
Comment (ezyang) [inline torch/csrc/distributed/c10d/Utils.hpp:317]: Consider making this just int64_t

**Review by ezyang — APPROVED** [review]

File: torch/csrc/distributed/c10d/reducer.hpp (hunk 1 above)
Comment (cyyever) [reply torch/csrc/distributed/c10d/reducer.hpp]: Fixed

File: torch/csrc/distributed/c10d/Utils.hpp (hunk 2 above)
Comment (cyyever) [reply torch/csrc/distributed/c10d/Utils.hpp]: Fixed

**Pull request closed without merging**